core/src/test/scala/quasar/destination/gbq/GBQDestinationModuleSpec.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2014–2019 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.destination.gbq
+
+import slamdata.Predef._
+
+import quasar.EffectfulQSpec
+import quasar.api.destination.{Destination, DestinationType, DestinationError}
+import quasar.api.destination.DestinationError.InitializationError
+import quasar.connector.ResourceError
+import quasar.contrib.scalaz.MonadError_
+
+import argonaut._, Argonaut._
+
+import cats.effect.{IO, Timer, ContextShift}
+
+import eu.timepit.refined.auto._
+
+import java.nio.file.{Files, Paths}
+import java.nio.charset.StandardCharsets.UTF_8
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object GBQDestinationModuleSpec extends EffectfulQSpec[IO] {
+
+   "initialization" should {
+     "fail with malformed config when config is not decodable" >>* {
+      val cfg = Json("malformed" := true)
+      dest(cfg)(r => IO.pure(r match {
```

Comment:
`pure` is not the correct thing to be using here. `must_===` is an effectful matcher (it throws when it fails), and is therefore not a value, and thus cannot be safely wrapped by `pure`. You should just be using `IO {`